app/src/main/java/ch/sbb/polarion/extension/generic/properties/ExtendedProperties.java

```diff
@@ -1,59 +1,42 @@
 package ch.sbb.polarion.extension.generic.properties;
 
+import lombok.EqualsAndHashCode;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 import java.util.HashMap;
-import java.util.Properties;
+import java.util.Set;
 
-public class ExtendedProperties extends Properties {
+@EqualsAndHashCode
+public class ExtendedProperties {
 
-    private final HashMap<Object, String> defaultValues = new HashMap<>();
-    private final HashMap<Object, String> descriptions = new HashMap<>();
-
-    public ExtendedProperties() {
-        super();
+    public record Value(@NotNull String value, @Nullable String defaultValue, @Nullable String description) {
     }
 
-    public ExtendedProperties(int size) {
-        super(size);
-    }
+    private final HashMap<String, Value> properties;
 
-    @Override
-    public synchronized Object setProperty(@NotNull String key, @NotNull String value) {
-        return this.setProperty(key, value, null, null);
-    }
-
-    public synchronized Object setProperty(@NotNull String key, @NotNull String value, @Nullable String defaultValue, @Nullable String description) {
-        return this.put(key, value, defaultValue, description);
+    public ExtendedProperties() {
+        this(8);
     }
 
-    public synchronized Object put(@NotNull Object key, @NotNull Object value, @Nullable String defaultValue, @Nullable String description) {
-        defaultValues.put(key, defaultValue);
-        descriptions.put(key, description);
-        return super.put(key, value);
+    public ExtendedProperties(int size) {
+        properties = new HashMap<>(size);
```

Comment:
Are you sure that call wouldn't be accessed concurrently? If it can, that ConcurrentHashMap is better choice

Reply:
Fixed